File: javascript/aws-lambda/security/tainted-html-response.js
exports.handler = function (event, context) {
    const html = `<div>${event.name}</div>`;

    const someRandomStuff = {
        // ok: tainted-html-response
        data: event.foo
    }
    bar(someRandomStuff)

    const response = {
        statusCode: 200,
        // ruleid: tainted-html-response
        body: html,
        headers: {
            'Content-Type': 'text/html',
        }
    };

    return response
}

File: python/aws-lambda/security/tainted-html-response.py
def lambda_handler(event, context):
	html = f"<div>{event['input']}</div>"

	foo = {
		# ok: tainted-html-response
		"data": event['foo']
	}
	bar(foo)

	result = {
		"statusCode": 200,
		# ruleid: tainted-html-response
		"body": html,
		"headers": {
			"Content-Type": "text/html"
		}
	}
	return result
